plugin/core/sessions.py

```diff
@@ -894,34 +899,43 @@ def m_window_workDoneProgress_create(self, params: Any, request_id: Any) -> None
         self._progress[params['token']] = dict()
         self.send_response(Response(request_id, None))
 
+    def _progress_status_key(self, token: str) -> str:
+        return "lspprogress{}{}".format(self.config.name, token)
+
     def m___progress(self, params: Any) -> None:
         """handles the $/progress notification"""
         token = params['token']
-        if token not in self._progress:
+        data = self._progress.get(token)
+        if not isinstance(data, dict):
             debug('unknown $/progress token: {}'.format(token))
             return
         value = params['value']
-        if value['kind'] == 'begin':
-            self._progress[token]['title'] = value['title']  # mandatory
-            self._progress[token]['message'] = value.get('message')  # optional
-            self.window.status_message(self._progress_string(token, value))
-        elif value['kind'] == 'report':
-            self.window.status_message(self._progress_string(token, value))
-        elif value['kind'] == 'end':
-            if value.get('message'):
-                status_msg = self._progress[token]['title'] + ': ' + value['message']
-                self.window.status_message(status_msg)
+        kind = value['kind']
+        key = self._progress_status_key(token)
+        if kind == 'begin':
+            data['title'] = value['title']  # mandatory
+            data['message'] = value.get('message')  # optional
+            progress_string = self._progress_string(data, value)
+            for sv in self.session_views_async():
+                sv.view.set_status(key, progress_string)
+        elif kind == 'report':
+            progress_string = self._progress_string(data, value)
+            for sv in self.session_views_async():
+                sv.view.set_status(key, progress_string)
+        elif kind == 'end':
+            for sv in self.session_views_async():
```

Comment:
I guess we are not on the right thread so this could cause issues?
But with #1348 it should be fine?

Reply:
> But with #1348 it should be fine?

Yes. I guess it's time to debug that now...